File: app/src/main/java/cloud/app/vvf/extension/builtIn/tvdb/SgTheTvdbInterceptor.java
package cloud.app.vvf.extension.builtIn.tvdb;

import com.uwetrottmann.thetvdb.TheTvdb;
import com.uwetrottmann.thetvdb.TheTvdbInterceptor;

import java.io.IOException;

import javax.inject.Inject;

import dagger.Lazy;
import okhttp3.Interceptor;
import okhttp3.Response;

/**
 * A custom {@link TheTvdbInterceptor} which does not require a {@link
 * TheTvdb} instance until intercepting.
 */
public class SgTheTvdbInterceptor implements Interceptor {

    private final Lazy<TheTvdb> theTvdb;

    @Inject
    public SgTheTvdbInterceptor(Lazy<TheTvdb> theTvdb) {
        this.theTvdb = theTvdb;
    }

    @Override
    public Response intercept(Chain chain) throws IOException {
        return TheTvdbInterceptor.handleIntercept(chain, theTvdb.get().jsonWebToken());
    }
}

File: app/src/main/java/cloud/app/vvf/extension/tmdb/services/tvdb/SgTheTvdbInterceptor.kt
package cloud.app.vvf.extension.tmdb.services.tvdb

import com.uwetrottmann.thetvdb.TheTvdb
import com.uwetrottmann.thetvdb.TheTvdbInterceptor
import okhttp3.Interceptor
import okhttp3.Response
import java.io.IOException

/**
 * A custom [TheTvdbInterceptor] which does not require a [ ] instance until intercepting.
 */
class SgTheTvdbInterceptor constructor(val theTvdb: TheTvdb) : Interceptor {

    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        return TheTvdbInterceptor.handleIntercept(chain, theTvdb.jsonWebToken())
    }
}
